### DISCUSSION TITLE
MotionBlur doesn't use OpenCL

### DISCUSSION THREAD

**Discussion by Mr1Penguin** [ImageMagick/ImageMagick#6700] [Contribute]
Hello!

Starting with 7 version of ImageMagick MotionBlur doesn't use OpenCl for acceleration. Is this supposed to be this way?

**Comment by dlemstra** [comment]
Thanks for reporting this. The call to the accelerated method was missing. I added the call but it turns out the kernel does not work with IM7. It needs to be fixed.

**Comment by Mr1Penguin** [comment]
I modified method MotionBlurImage(const Image *, ...) in downloaded sources for Visual Studio. I took some code from IM6 version. To make it work I add alpha channel if image does not have it.
https://gist.github.com/Mr1Penguin/f164fab832476d90b311df6d30a0f6a7#file-effects-c-L63-L72

**Comment by dlemstra** [comment]
Thanks for replying, now I know what was wrong with the call that I added. The kernel works properly but only when the image has an alpha channel. The OpenCL kernel only supports RGBA images at the moment. We will need to modify the kernel and add support for this. Will leave this issue open so we won't forget to do that.
